tap_hubspot_beta/tap.py

```diff
@@ -187,6 +197,53 @@ def catalog_dict(self) -> dict:
                         stream["schema"]["properties"][field]["field_meta"] = stream_class.fields_metadata.get(field, {})
         return catalog
 
+    def generate_stream_class(self, custom_object: Dict[str, Any]) -> hubspotV3Stream:
+        # check for required fields to construct the custom objects class
+        _id = custom_object.get("id")
+        if not _id:
+            raise ValueError(f"Missing id in custom object. object={custom_object}.")
+        name = custom_object.get("name")
+        if not name:
+            raise ValueError(f"Missing name in custom object. id={_id}.")
+        object_type_id = custom_object.get("objectTypeId")
+        if not object_type_id:
+            raise ValueError(f"Missing objectTypeId in custom object. id={_id}. name={name}")
+        properties = custom_object.get("properties")
+        if not properties:
+            raise ValueError(f"Missing properties in custom object. id={_id}. name={name}. object_type_id={object_type_id}")
+        
```

Comment:
Can you check every property and raise a unique Value error with the list of missing properties?

Reply:
done, thank you for the call